src/Microsoft.AspNet.Mvc.Core/Filters/AuthenticateAttribute.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Mvc
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public class AuthenticateAttribute :
+        Attribute, IAsyncAuthenticationFilter, IAuthenticationFilter
+    {
+        private string _authTypes;
+        private string[] _authTypesSplit;
+
+        public AuthenticateAttribute() { }
+
+        public AuthenticateAttribute(string authenticationTypes)
```

Comment:
Out of curiosity, why not directly accepting a `params string[]` parameter? (kill magic strings with :fire:!)

```csharp
[Authenticate(
    CookieAuthenticationDefaults.AuthenticationType,
    OAuthBearerAuthenticationDefaults.AuthenticationType)]
```